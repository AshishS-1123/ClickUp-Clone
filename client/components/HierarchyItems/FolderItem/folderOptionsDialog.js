import React, { useState } from "react";
import AddIcon from '@mui/icons-material/Add';
import DeleteIcon from '@mui/icons-material/Delete';
import EditIcon from '@mui/icons-material/Edit';
import Menu from "@mui/material/Menu";
import MenuItem from "@mui/material/MenuItem";
import CreateListDialog from "../../Dialogs/createListDialog";
import CreateFolderDialog from "../../Dialogs/CreateFolderDialog";
import CreateTaskDialog from "../../Dialogs/createTaskDialog";

function FolderOptionsDialog({ anchorEl, openMenu, closeMenu, itemId, itemType }) {
  const [showListDialog, setShowListDialog] = useState(false);
  const [showFolderDialog, setShowFolderDialog] = useState(false);
  const [showTaskDialog, setShowTaskDialog] = useState(false);

  const menuItemStyle = {
    padding: "8px",
    color: "#d5d6d7",
    fontSize: "13px",
  }

  const iconStyle = {
    marginRight: "5px",
    transform: "scale(0.8)",
  }

  return (
    <>
      <Menu
        sx={{
          "& .MuiMenu-paper": {
            background: "#384047",
            color: "#979797",
          },
          "& li": {
            fontSize: "13px",
          }
        }}
        anchorEl={anchorEl}
        open={openMenu}
        onClose={closeMenu}
      >
        <h5 style={{ fontWeight: "600", padding: "5px 20px", fontSize: "11px", color: "#dadada" }}>FOLDER SETTINGS</h5>
        <MenuItem style={menuItemStyle} onClick={() => { setShowFolderDialog(true) }}><AddIcon style={iconStyle} />Add Folder</MenuItem>
        <MenuItem style={menuItemStyle} onClick={() => { setShowListDialog(true) }}><AddIcon style={iconStyle} />Add List</MenuItem>
        <MenuItem style={menuItemStyle} onClick={() => { setShowTaskDialog(true) }}><AddIcon style={iconStyle} />Add Task</MenuItem>
        <MenuItem style={menuItemStyle}><EditIcon style={iconStyle} />Edit</MenuItem>
        <MenuItem style={menuItemStyle}><DeleteIcon style={iconStyle} />Delete</MenuItem>
      </Menu>

      <CreateFolderDialog
        open={showFolderDialog}
        closeDialog={() => { setShowFolderDialog(false); closeMenu() }}
        itemType={itemType}
        itemId={itemId}
      />
      <CreateListDialog
        open={showListDialog}
        closeDialog={() => { setShowListDialog(false); closeMenu() }}
        itemType={itemType}
        itemId={itemId}
      />
      <CreateTaskDialog
        open={showTaskDialog}
        closeDialog={() => { setShowTaskDialog(false); closeMenu() }}
        itemType={itemType}
        itemId={itemId}
      />

    </>
  )
}

export default FolderOptionsDialog;
