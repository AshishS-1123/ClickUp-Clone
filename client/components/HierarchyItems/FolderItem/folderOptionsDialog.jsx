import React, { useState } from "react";
import AddIcon from "@mui/icons-material/Add";
import DeleteIcon from "@mui/icons-material/Delete";
import EditIcon from "@mui/icons-material/Edit";
import Menu from "@mui/material/Menu";
import MenuItem from "@mui/material/MenuItem";
import CreateListDialog from "../../Dialogs/CreateListDialog";
import CreateFolderDialog from "../../Dialogs/CreateFolderDialog";
import CreateTaskDialog from "../../Dialogs/CreateTaskDialog";
import themeColors from "../../../utils/contexts/themeContext";

function FolderOptionsDialog({
  anchorEl, openMenu, closeMenu, itemId, itemType,
}) {
  const [showListDialog, setShowListDialog] = useState(false);
  const [showFolderDialog, setShowFolderDialog] = useState(false);
  const [showTaskDialog, setShowTaskDialog] = useState(false);

  const menuItemStyle = {
    padding: "8px",
    color: themeColors.textColor,
    fontSize: "13px",
  };

  const iconStyle = {
    marginRight: "5px",
    transform: "scale(0.8)",
    color: "grey",
  };

  return (
    <>
      <Menu
        sx={{
          "& .MuiMenu-paper": {
            background: themeColors.background,
            color: themeColors.textColor,
          },
          "& li": {
            fontSize: "13px",
          },
        }}
        anchorEl={anchorEl}
        open={openMenu}
        onClose={closeMenu}
      >
        <h5 style={{
          fontWeight: 300, padding: "5px 20px", fontSize: "11px", color: themeColors.textBoldColor,
        }}
        >
          FOLDER SETTINGS
        </h5>
        <MenuItem style={menuItemStyle} onClick={() => { setShowFolderDialog(true); }}>
          <AddIcon style={iconStyle} />
          Add Folder
        </MenuItem>
        <MenuItem style={menuItemStyle} onClick={() => { setShowListDialog(true); }}>
          <AddIcon style={iconStyle} />
          Add List
        </MenuItem>
        <MenuItem style={menuItemStyle}>
          <EditIcon style={iconStyle} />
          Edit
        </MenuItem>
        <MenuItem style={menuItemStyle}>
          <DeleteIcon style={iconStyle} />
          Delete
        </MenuItem>
      </Menu>

      <CreateFolderDialog
        open={showFolderDialog}
        closeDialog={() => { setShowFolderDialog(false); closeMenu(); }}
        itemType={itemType}
        itemId={itemId}
      />
      <CreateListDialog
        open={showListDialog}
        closeDialog={() => { setShowListDialog(false); closeMenu(); }}
        itemType={itemType}
        itemId={itemId}
      />
    </>
  );
}

export default FolderOptionsDialog;
