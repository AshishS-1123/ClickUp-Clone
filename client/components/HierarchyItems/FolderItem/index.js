import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import styles from "../SpaceItem/SpaceItem.module.css";
import FolderIcon from '@mui/icons-material/Folder';
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
import MoreHorizIcon from '@mui/icons-material/MoreHoriz';
import ListItem from "../ListItem";
import { setActive } from "../../../redux/slices/spaceSlice";
import FolderOptionsDialog from "./folderOptionsDialog";

function FolderItem({ id, folderName, contents, nestingLevel = 10 }) {
  const [revealerVisible, setRevealerVisible] = useState(false);
  const [showIcons, setShowIcons] = useState(false);
  const isActive = useSelector(state => state.spaceReducer.activeItem) == id;
  const dispatch = useDispatch();
  const [anchorEl, setAnchorEl] = useState(null);
  const openMenu = Boolean(anchorEl);

  let containerStyle = {
    paddingLeft: `${nestingLevel}px`,
    color: "white",
  };

  if (isActive) {
    containerStyle = {
      ...containerStyle,
      background: "#3c3d39",
      borderLeft: "3px solid #ffa12f"
    }
  }

  const showRevealer = (event) => {
    setRevealerVisible(prev => !prev);
  }

  const setCurrentAsActive = (event) => {
    dispatch(setActive({ id }));
  }

  const showOptions = (event) => {
    setAnchorEl(event.currentTarget);
  }

  const closeOptions = (event) => {
    setAnchorEl(null);
  }

  return (
    <>
      <div
        className={styles.item__container}
        style={containerStyle}
        onMouseEnter={() => { setShowIcons(true) }}
        onMouseLeave={() => { setShowIcons(false) }}
      >

        <div className={styles.item__titleContainer} onClick={setCurrentAsActive}>
          <FolderIcon sx={{ width: "16px", height: "16px", color: "lightgrey" }} />
          <div className={styles.folder__title}>{folderName}</div>
        </div>

        <MoreHorizIcon
          onClick={showOptions}
          sx={{
            visibility: (showIcons || openMenu) ? "visible" : "hidden",
            transform: "scale(0.7)"
          }}
        />

        <KeyboardArrowDownIcon
          onClick={showRevealer}
          sx={{
            transform: revealerVisible ? "rotate(180deg)" : "",
            color: "white",
            display: showIcons ? "block" : "none",
          }}
        />

      </div>

      <div style={{ display: revealerVisible ? "block" : "none" }}>
        {
          contents.map(item => {
            if (item.itemType == "FOLDER") {
              return <FolderItem folderName={item.name} key={item.id} id={item.id} nestingLevel={nestingLevel + 10} />
            } else {
              return <ListItem listName={item.name} key={item.id} id={item.id} nestingLevel={nestingLevel + 10} />
            }
          })
        }
      </div>

      <FolderOptionsDialog
        anchorEl={anchorEl}
        openMenu={openMenu}
        closeMenu={closeOptions}
        itemId={id}
        itemType="folder"
      />
    </>
  )
}

export default FolderItem;
