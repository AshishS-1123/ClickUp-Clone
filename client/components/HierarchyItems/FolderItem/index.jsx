import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import FolderIcon from "@mui/icons-material/Folder";
import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
import MoreHorizIcon from "@mui/icons-material/MoreHoriz";
import styles from "../SpaceItem/SpaceItem.module.css";
import ListItem from "../ListItem";
import { setActive } from "../../../redux/slices/spaceSlice";
import FolderOptionsDialog from "./folderOptionsDialog";
import themeColors from "../../../utils/contexts/themeContext";

function FolderItem({
  id, folderName, contents, nestingLevel = 10,
}) {
  const [revealerVisible, setRevealerVisible] = useState(false);
  const [showIcons, setShowIcons] = useState(false);
  const isActive = useSelector((state) => state.spaceReducer.activeItem) == id;
  const dispatch = useDispatch();
  const [anchorEl, setAnchorEl] = useState(null);
  const openMenu = Boolean(anchorEl);

  let containerStyle = {
    paddingLeft: `${nestingLevel}px`,
    color: themeColors.textBoldColor,
  };

  if (isActive) {
    containerStyle = {
      ...containerStyle,
      background: themeColors.accentColorFaded,
      borderLeft: `3px solid ${themeColors.accentColor}`,
    };
  }

  const showRevealer = (event) => {
    setRevealerVisible((prev) => !prev);
  };

  const setCurrentAsActive = (event) => {
    dispatch(setActive({ id }));
  };

  const showOptions = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const closeOptions = (event) => {
    setAnchorEl(null);
  };

  return (
    <>
      <div
        className={styles.item__container}
        style={containerStyle}
        onMouseEnter={() => { setShowIcons(true); }}
        onMouseMove={() => { setShowIcons(true); }}
        onMouseLeave={() => { setShowIcons(false); }}
      >

        <div className={styles.item__titleContainer} onClick={setCurrentAsActive}>
          <FolderIcon sx={{ width: "16px", height: "16px", color: themeColors.textBoldColor }} />
          <h5 className={styles.folder__title}>{folderName}</h5>
        </div>

        <div style={{ display: "flex", flexDirection: "row" }}>
          <MoreHorizIcon
            onClick={showOptions}
            sx={{
              visibility: (showIcons || openMenu) ? "visible" : "hidden",
              transform: "scale(0.7)",
            }}
          />

          <KeyboardArrowDownIcon
            onClick={showRevealer}
            sx={{
              transform: revealerVisible ? "rotate(180deg) scale(0.7)" : "scale(0.7)",
              display: showIcons ? "block" : "none",
            }}
          />
        </div>

      </div>

      <div style={{ display: revealerVisible ? "block" : "none" }}>
        {
          contents && contents.map((item) => {
            if (item.itemType === "FOLDER") {
              return (
                <FolderItem
                  key={item.id}
                  folderName={item.name}
                  id={item.id}
                  nestingLevel={nestingLevel + 10}
                  contents={item.contents}
                />
              );
            } if (item.itemType === "LIST") {
              return (
                <ListItem
                  key={item.id}
                  id={item.id}
                  listName={item.name}
                  nestingLevel={nestingLevel + 10}
                />
              );
            }
          })
        }
      </div>

      <FolderOptionsDialog
        anchorEl={anchorEl}
        openMenu={openMenu}
        closeMenu={closeOptions}
        itemId={id}
        itemType="folder"
      />
    </>
  );
}

export default FolderItem;
