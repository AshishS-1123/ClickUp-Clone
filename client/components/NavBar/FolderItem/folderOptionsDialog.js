import React, { useState } from "react";
import AddIcon from '@mui/icons-material/Add';
import DeleteIcon from '@mui/icons-material/Delete';
import EditIcon from '@mui/icons-material/Edit';
import Menu from "@mui/material/Menu";
import MenuItem from "@mui/material/MenuItem";
import CreateListDialog from "../../Dialogs/createListDialog";

function FolderOptionsDialog({ anchorEl, openMenu, closeMenu }) {
  const [showListDialog, setShowListDialog] = useState(false);

  const menuItemStyle = {
    padding: "8px",
    color: "#d5d6d7",
    fontSize: "13px",
  }

  return (
    <>
      <Menu
        sx={{
          "& .MuiMenu-paper": {
            background: "#384047",
            color: "#979797",
          },
          "& li": {
            fontSize: "13px",
          }
        }}
        anchorEl={anchorEl}
        open={openMenu}
        onClose={closeMenu}
      >
        <h5 style={{ fontWeight: "600", padding: "5px 20px", fontSize: "11px" }}>FOLDER SETTINGS</h5>
        <MenuItem style={menuItemStyle}><AddIcon />Add Folder</MenuItem>
        <MenuItem style={menuItemStyle} onClick={() => { setShowListDialog(true) }}><AddIcon />Add List</MenuItem>
        <MenuItem style={menuItemStyle}><EditIcon />Edit</MenuItem>
        <MenuItem style={menuItemStyle}><DeleteIcon />Delete</MenuItem>
      </Menu>

      <CreateListDialog open={showListDialog} closeDialog={() => { setShowListDialog(false) }} />
    </>
  )
}

export default FolderOptionsDialog;
