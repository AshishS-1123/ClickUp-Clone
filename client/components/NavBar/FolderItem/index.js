import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import styles from "../SpaceItem/SpaceItem.module.css";
import FolderIcon from '@mui/icons-material/Folder';
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
import ListItem from "../ListItem";
import { setActive } from "../../../redux/slices/spaceSlice";

function FolderItem({ id, folderName, contents, nestingLevel = 10 }) {
  const [revealerVisible, setRevealerVisible] = useState(false);
  const [showIcons, setShowIcons] = useState(false);
  const isActive = useSelector(state => state.spaceReducer.activeItem) == id;
  const dispatch = useDispatch();

  let containerStyle = {
    paddingLeft: `${nestingLevel}px`,
    color: "white",
  };

  if (isActive) {
    console.log("Folder active");
    containerStyle = {
      ...containerStyle,
      background: "#3c3d39",
      borderLeft: "3px solid #ffa12f"
    }
  }

  const showRevealer = (event) => {
    setRevealerVisible(prev => !prev);
  }

  const setCurrentAsActive = (event) => {
    dispatch(setActive({ id }));
  }

  return (
    <>
      <div
        className={styles.item__container}
        style={containerStyle}
        onMouseEnter={() => { setShowIcons(true) }}
        onMouseLeave={() => { setShowIcons(false) }}
      >

        <div className={styles.item__titleContainer} onClick={setCurrentAsActive}>
          <FolderIcon sx={{ width: "16px", height: "16px", color: "lightgrey" }} />
          <div className={styles.folder__title}>{folderName}</div>
        </div>

        <KeyboardArrowDownIcon
          onClick={showRevealer}
          sx={{
            transform: revealerVisible ? "rotate(180deg)" : "",
            color: "white",
            display: showIcons ? "block" : "none",
          }}
        />

      </div>

      <div style={{ display: revealerVisible ? "block" : "none" }}>
        {
          contents.map(item => {
            if (item.itemType == "FOLDER") {
              return <FolderItem folderName={item.name} key={item.id} id={item.id} nestingLevel={nestingLevel + 10} />
            } else {
              return <ListItem listName={item.name} key={item.id} id={item.id} nestingLevel={nestingLevel + 10} />
            }
          })
        }
      </div>
    </>
  )
}

export default FolderItem;
