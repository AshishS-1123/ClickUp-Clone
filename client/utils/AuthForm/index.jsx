import React, { useRef } from "react";
import Link from "next/link";
import { useDispatch, useSelector } from "react-redux";
import styles from "./AuthForm.module.css";
import protectedRoute from "../ProtectedRoutes";
import PropTypes from "prop-types";

const dataFromAuthType = (authType) => {
  const metaData = {};

  if (authType === "login") {
    metaData.formHeading = "Welcome back!";
    metaData.alternateAuthLink = "/register";
    metaData.alternateAuthText = "Don't have an account? Register";
  } else if (authType === "register") {
    metaData.formHeading = "Let's go!";
    metaData.alternateAuthLink = "/login";
    metaData.alternateAuthText = "Already have an account? Log In";
  } else {
    metaData.formHeading = "";
    metaData.alternateAuthLink = "";
    metaData.alternateAuthText = "";
  }

  return metaData;
};

function AuthForm({ reducer, authType }) {
  const emailRef = useRef(null);
  const passwordRef = useRef(null);
  const dispatch = useDispatch();
  const error = useSelector((state) => state.authReducer.error);

  const { formHeading, alternateAuthLink, alternateAuthText } = dataFromAuthType(authType);
  const clickHandler = (event) => {
    event.preventDefault();

    const userEmail = emailRef.current.value;
    const password = passwordRef.current.value;

    dispatch(reducer({ userEmail, password }))
  };

  return (
    <form className={styles.auth_form}>
      <h1 className={styles.auth_heading}>{formHeading}</h1>

      <div className={styles.input_container}>
        <label htmlFor="auth__email" className={styles.auth_label}>Email</label>
        <input ref={emailRef} type="text" id="auth__email" className={styles.auth_input} placeholder="example@site.com" />
      </div>

      <div className={styles.input_container}>
        <label htmlFor="auth__password" className={styles.auth_label}>Password</label>
        <input ref={passwordRef} type="password" id="auth__password" className={styles.auth_input} placeholder="*****" />
      </div>

      <p className={styles.error_message}>
        {error}
      </p>

      <button type="submit" className={styles.auth_button} onClick={clickHandler}>Start playing</button>

      <Link href={alternateAuthLink}>
        <a className={styles.auth_link}>{alternateAuthText}</a>
      </Link>
    </form>
  );
}

AuthForm.propTypes = {
  reducer: PropTypes.func.isRequired,
  authType: PropTypes.string.isRequired,
};

export default protectedRoute(AuthForm, false);
