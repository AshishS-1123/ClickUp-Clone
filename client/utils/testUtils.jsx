import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit"
import { Provider } from "react-redux";

import authReducer from "../redux/slices/authSlice";
import workspaceReducer from "../redux/slices/workspaceSlice";
import spaceReducer from "../redux/slices/spaceSlice";


const reducers = {
  authReducer,
  workspaceReducer,
  spaceReducer,
}

function render(
  ui,
  {
    preloadedState,
    store = configureStore({ reducer: reducers, preloadedState }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from "@testing-library/react"
// override render method
export { render }
