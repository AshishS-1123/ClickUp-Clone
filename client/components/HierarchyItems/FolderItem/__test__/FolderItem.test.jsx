import React from "react";
import { render, screen, cleanup, waitFor, fireEvent, within } from "../../../../utils/testUtils";
import userEvent from "@testing-library/user-event";
import FolderItem from "../../FolderItem";

afterEach(cleanup);

describe("<FolderItem /> tests", () => {
  it("should match snapshot", () => {
    const component = render(
      <FolderItem id="1" folderName="Folder 1" contents={[]} nestingLevel={10} />
    );

    expect(component.container).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    const component = render(
      <FolderItem id="1" folderName="Folder 1" contents={[]} nestingLevel={10} />
    );
  });

  it("should update redux store when marked as active", () => {
  });

  it("should show options button and arrow when hovered", async () => {
    const component = render(
      <FolderItem id="1" folderName="Folder 1" contents={[]} nestingLevel={10} />
    );

    // Icons should not be shown before hover.
    expect(screen.queryByTestId("MoreHorizIcon")).not.toBeVisible();
    expect(screen.queryByTestId("KeyboardArrowDownIcon")).not.toBeVisible();

    fireEvent.mouseOver(component.container);

    await waitFor(() => {
      expect(screen.getByTestId("MoreHorizIcon")).toHaveStyle("display: inline-block");
      expect(screen.getByTestId("KeyboardArrowDownIcon")).toHaveStyle("visibility: visible");
    })
  });

})
