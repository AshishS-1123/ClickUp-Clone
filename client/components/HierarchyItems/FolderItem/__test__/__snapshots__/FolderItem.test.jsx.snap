// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FolderItem /> tests should match snapshot 1`] = `
<div>
  <div
    class="item__container"
    style="padding-left: 10px; color: white;"
  >
    <div
      class="item__titleContainer"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-136s91u-MuiSvgIcon-root"
        data-testid="FolderIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
        />
      </svg>
      <h5
        class="folder__title"
      >
        Folder 1
      </h5>
    </div>
    <div
      style="display: flex; flex-direction: row;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-78bnp2-MuiSvgIcon-root"
        data-testid="MoreHorizIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-z7a6zs-MuiSvgIcon-root"
        data-testid="KeyboardArrowDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
    </div>
  </div>
  <div
    style="display: none;"
  />
</div>
`;
